src/dot_parser/mod.rs

```diff
@@ -0,0 +1,316 @@
+use pest::Parser;
+use pest_derive::Parser;
+use pyo3::prelude::*;
+use pyo3::types::{PyDict, PyString};
+
+use crate::digraph::PyDiGraph;
+use crate::graph::PyGraph;
+use crate::StablePyGraph;
+
+use rustworkx_core::petgraph::prelude::{Directed, NodeIndex, Undirected};
+use std::collections::HashMap;
```

Comment:
Use the hashmap from hashbrown: https://docs.rs/hashbrown/latest/hashbrown/struct.HashMap.html

That's what we use everywhere